grype/version/format.go

```diff
@@ -11,6 +11,7 @@ const (
 	SemanticFormat
 	ApkFormat
 	DebFormat
+	JavaFormat
```

Comment:
I think this should be oriented around maven ecosystem instead of the higher order java language based on where the spec is https://maven.apache.org/pom.html#Version_Order_Specification

Reply:
I've reoriented the naming of the additions in this PR around `maven` given that they are the authors of the version order specification we're now following for packages with type `JavaPkg` going forward.

I think you also mentioned a change for https://github.com/anchore/grype-db where we should start labeling these constraints correctly?